pdm/cli/completions/pdm.zsh

```diff
@@ -1,6 +1,7 @@
 #compdef pdm
 
 PDM_PIP_INDEXES=($(command pdm config pypi.url))
+PYTHON="%{python_executable}"
```

Comment:
As a global variable, `PYTHON` is a too common name. Users may use this name for their own plugin.
I suggest something like `PDM_PYTHON` so that it won't be override/access by other script.